detector.go

```diff
@@ -35,7 +35,7 @@ type BuildPlanEntry struct {
 func (be BuildPlanEntry) noOpt() BuildPlanEntry {
 	var out []Buildpack
 	for _, p := range be.Providers {
-		out = append(out, p.noOpt())
+		out = append(out, p.noOpt().noAPI())
```

Comment:
I've seen `noAPI` a few times now in this PR. I think I understand this is wiping it out so it won't get output to disk somewhere. Would it make sense to have a buildplan.buildpack writer that ignores that key or something?

Reply:
Because `noOpt()` is doing a similar thing for a similar reasons (in some places, like the one above) I figured I would follow the existing pattern. I tried a different approach but things got gnarly fast and I determined it would required a larger refactor (I didn't want to make this already large PR larger)